src/routes/invidious_routes/download.ts

```diff
@@ -0,0 +1,95 @@
+import type { Context, Hono } from "hono";
+import { z } from "zod";
+import { HTTPException } from "hono/http-exception";
+import { verifyRequest } from "../../lib/helpers/verifyRequest.ts";
+
+const DownloadWidgetSchema = z.discriminatedUnion("ext", [
```

Comment:
this throws an error on launch. I believe the discriminate union needs literals to work properly, because it can determine the type entirely through those literals. 

In this case, I think you can just do `.or` to join the two